### PR TITLE
[New Rule] Hosts File Modified

### LINKED ISSUE #24: [New Rule] Hosts File Modified
## Description
The Host file on an endpoint is used to control manual IP to hostname resolutions. The Host file is the first point of lookup for DNS hostname resolution so if adversaries can modify the endpoint Host file, they can route traffic to malicious infrastructure. This rule detects modifications to the Host file on Linux (Ubuntu or RHEL), macOS, and Windows systems.

## Required Info

- **Eventing Sources:**
Auditbeat

- **Target Operating Systems:**
Linux (Ubuntu & RHEL), macOS, and Windows

- **Platforms**
NA

- **Target ECS Version:** 1.5.0
- **New fields required in ECS for this? Associated issue/PR:** NA

## Optional Info
- **References:**

## Example Data
image
image

CentOS
```
{
  "_index": "auditbeat-7.8.0-2020.07.07",
  "_type": "_doc",
  "_id": "JxmqKXMBpYSg_Jq8_rEZ",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-07-07T14:25:29.237Z",
    "host": {
      "id": "b97c1797f883eb5f3d72134d451d0384",
      "mac": [
        "42:01:0a:80:00:1e",
        "42:01:0a:0a:08:02"
      ],
      "architecture": "x86_64",
      "name": "[redacted]",
      "hostname": "[redacted]",
      "os": {
        "codename": "Core",
        "platform": "centos",
        "version": "7 (Core)",
        "name": "CentOS Linux",
        "family": "redhat",
        "kernel": "3.10.0-1127.13.1.el7.x86_64"
      },
      "containerized": false,
      "ip": [
        "10.10.8.2"
      ]
    },
    "@version": "1",
    "file": {
      "path": "/etc/hosts"
    },
    "fields": {
      "labels": {
        "responsible_human": "[redacted]"
      }
    },
    "service": {
      "type": "file_integrity"
    },
    "agent": {
      "type": "auditbeat",
      "ephemeral_id": "06db1f12-0606-4e12-9044-752bf0fdb95c",
      "version": "7.8.0",
      "name": "[redacted]",
      "hostname": "[redacted]",
      "id": "24c3c24f-4114-497b-8148-f0189407029c"
    },
    "ecs": {
      "version": "1.5.0"
    },
    "event": {
      "type": [
        "change"
      ],
      "kind": "event",
      "category": [
        "file"
      ],
      "dataset": "file",
      "module": "file_integrity",
      "action": [
        "moved"
      ]
    },
    "tags": [
      "beats_input_raw_event"
    ]
  },
  "fields": {
    "file.created": [],
    "system.audit.package.installtime": [],
    "process.parent.start": [],
    "event.end": [],
    "file.ctime": [],
    "tls.client.not_after": [],
    "system.audit.user.password.last_changed": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-07-07T14:25:29.237Z"
    ],
    "file.mtime": [],
    "event.created": [],
    "file.accessed": [],
    "system.audit.host.boottime": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "event.start": [],
    "tls.server.not_after": [],
    "process.start": []
  },
  "highlight": {
    "file.path": [
      "@kibana-highlighted-field@/etc/hosts@/kibana-highlighted-field@"
    ],
    "event.category": [
      "@kibana-highlighted-field@file@/kibana-highlighted-field@"
    ],
    "event.type": [
      "@kibana-highlighted-field@change@/kibana-highlighted-field@"
    ],
    "agent.version": [
      "@kibana-highlighted-field@7.8.0@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1594131929237
  ]
}
```
Ubuntu
```
{
  "_index": "auditbeat-7.8.0-2020.06.18-000001",
  "_type": "_doc",
  "_id": "T_OoKXMBrrfU-yfro_do",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-07-07T14:23:03.446Z",
    "cloud": {
      "machine": {
        "type": "n1-standard-1"
      },
      "project": {
        "id": "[redacted]"
      },
      "provider": "gcp",
      "availability_zone": "us-central1-a",
      "instance": {
        "name": "[redacted]",
        "id": "5070623776091015145"
      }
    },
    "hash": {
      "sha1": "123a15a4b4abca52cb9da17574fe67b519154cbc"
    },
    "file": {
      "owner": "root",
      "size": 279,
      "uid": "0",
      "mode": "0644",
      "mtime": "2020-07-07T14:23:03.433Z",
      "inode": "13920",
      "gid": "0",
      "path": "/etc/hosts",
      "ctime": "2020-07-07T14:23:03.437Z",
      "type": "file",
      "group": "root",
      "hash": {
        "sha1": "123a15a4b4abca52cb9da17574fe67b519154cbc"
      }
    },
    "event": {
      "module": "file_integrity",
      "dataset": "file",
      "kind": "event",
      "category": [
        "file"
      ],
      "type": [
        "change"
      ],
      "action": [
        "moved"
      ]
    },
    "service": {
      "type": "file_integrity"
    },
    "host": {
      "hostname": "[redacted]",
      "architecture": "x86_64",
      "os": {
        "family": "debian",
        "name": "Ubuntu",
        "kernel": "4.15.0-1036-gcp",
        "codename": "bionic",
        "platform": "ubuntu",
        "version": "18.04.4 LTS (Bionic Beaver)"
      },
      "id": "d34db77d59e4bff23ff418db1336a6a6",
      "name": "[redacted]",
      "containerized": false,
      "ip": [
        "10.128.0.13",
        "fe80::4001:aff:fe80:d",
        "fe80::38ae:4aff:fe00:b967"
      ],
      "mac": [
        "42:01:0a:80:00:0d",
        "3a:ae:4a:00:b9:67"
      ]
    },
    "agent": {
      "hostname": "[redacted]",
      "ephemeral_id": "65aab5cb-52e6-4078-8ab8-a2501b914f0a",
      "id": "a9619787-11ba-4f97-a476-f0f20aaa286d",
      "name": "[redacted]",
      "type": "auditbeat",
      "version": "7.8.0"
    },
    "ecs": {
      "version": "1.5.0"
    }
  },
  "fields": {
    "file.created": [],
    "system.audit.package.installtime": [],
    "process.parent.start": [],
    "event.end": [],
    "file.ctime": [
      "2020-07-07T14:23:03.437Z"
    ],
    "tls.client.not_after": [],
    "system.audit.user.password.last_changed": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-07-07T14:23:03.446Z"
    ],
    "file.mtime": [
      "2020-07-07T14:23:03.433Z"
    ],
    "event.created": [],
    "file.accessed": [],
    "system.audit.host.boottime": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "event.start": [],
    "tls.server.not_after": [],
    "process.start": []
  },
  "highlight": {
    "file.path": [
      "@kibana-highlighted-field@/etc/hosts@/kibana-highlighted-field@"
    ],
    "event.category": [
      "@kibana-highlighted-field@file@/kibana-highlighted-field@"
    ],
    "event.type": [
      "@kibana-highlighted-field@change@/kibana-highlighted-field@"
    ],
    "agent.version": [
      "@kibana-highlighted-field@7.8.0@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1594131783446
  ]
}
```
macOS
```
{
  "_index": "auditbeat-7.8.0-2020.06.18-000001",
  "_type": "_doc",
  "_id": "cucsC3MBpYSg_Jq8tud0",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-07-01T16:19:05.374Z",
    "agent": {
      "id": "5d637140-43d7-4703-a184-14cf0935acf3",
      "name": "[redacted]",
      "type": "auditbeat",
      "version": "7.8.0",
      "hostname": "[redacted].lan",
      "ephemeral_id": "9c86abf0-2bc1-421a-b700-755030402aaa"
    },
    "hash": {
      "sha1": "774e692ef2d4b361a83645163b855c953c9eb784"
    },
    "event": {
      "action": [
        "moved"
      ],
      "module": "file_integrity",
      "dataset": "file",
      "kind": "event",
      "category": [
        "file"
      ],
      "type": [
        "change"
      ]
    },
    "service": {
      "type": "file_integrity"
    },
    "file": {
      "type": "file",
      "gid": "0",
      "hash": {
        "sha1": "774e692ef2d4b361a83645163b855c953c9eb784"
      },
      "inode": "18430329",
      "ctime": "2020-07-01T16:19:05.351Z",
      "size": 440,
      "owner": "root",
      "path": "/private/etc/hosts",
      "mode": "0644",
      "mtime": "2020-07-01T16:19:05.351Z",
      "group": "wheel",
      "uid": "0"
    },
    "ecs": {
      "version": "1.5.0"
    },
    "host": {
      "hostname": "[redacted].local",
      "architecture": "x86_64",
      "os": {
        "kernel": "19.5.0",
        "build": "19F101",
        "platform": "darwin",
        "version": "10.15.5",
        "family": "darwin",
        "name": "Mac OS X"
      },
      "id": "7A1239AF-8E8B-5928-A112-24FDC1718CF9",
      "name": "[redacted]",
      "ip": [
        "fe80::aede:48ff:fe00:1122",
        "192.168.158.141",
        "fe80::8897:16ff:fe26:4adc",
        "fe80::8897:16ff:fe26:4adc",
        "fe80::7a58:640d:54c9:582b",
        "fe80::1c29:ae02:3755:4761",
        "fe80::95e2:9d1:58c1:3663",
        "fe80::5c66:70c2:b21c:8e2a",
        "fe80::e87c:124a:867d:d757",
        "fe80::3b40:c943:ef9c:e9a6"
      ],
      "mac": [
        "ac:de:48:00:11:22",
        "a6:83:e7:ac:3c:98",
        "a4:83:e7:ac:3c:98",
        "82:ae:16:c5:30:01",
        "82:ae:16:c5:30:00",
        "82:ae:16:c5:30:05",
        "82:ae:16:c5:30:04",
        "82:ae:16:c5:30:01",
        "06:83:e7:ac:3c:98",
        "8a:97:16:26:4a:dc",
        "8a:97:16:26:4a:dc"
      ]
    }
  },
  "fields": {
    "file.created": [],
    "system.audit.package.installtime": [],
    "process.parent.start": [],
    "event.end": [],
    "file.ctime": [
      "2020-07-01T16:19:05.351Z"
    ],
    "tls.client.not_after": [],
    "system.audit.user.password.last_changed": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-07-01T16:19:05.374Z"
    ],
    "file.mtime": [
      "2020-07-01T16:19:05.351Z"
    ],
    "event.created": [],
    "file.accessed": [],
    "system.audit.host.boottime": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "event.start": [],
    "tls.server.not_after": [],
    "process.start": []
  },
  "highlight": {
    "file.path": [
      "@kibana-highlighted-field@/private/etc/hosts@/kibana-highlighted-field@"
    ],
    "event.category": [
      "@kibana-highlighted-field@file@/kibana-highlighted-field@"
    ],
    "event.type": [
      "@kibana-highlighted-field@change@/kibana-highlighted-field@"
    ],
    "agent.version": [
      "@kibana-highlighted-field@7.8.0@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1593620345374
  ]
}
```
Windows
```
{
  "_index": "auditbeat-7.8.0-2020.06.18-000001",
  "_type": "_doc",
  "_id": "SpTxzHIBrrfU-yfroJmi",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2020-06-19T14:18:05.780Z",
    "host": {
      "id": "5907f390-9809-47ea-973c-1f4202199ba6",
      "ip": [
        "fe80::f162:c81d:49cb:8aa0",
        "10.0.2.15"
      ],
      "mac": [
        "08:00:27:b7:f8:e7"
      ],
      "hostname": "DESKTOP-[redacted]",
      "architecture": "x86_64",
      "name": "DESKTOP-[redacted]",
      "os": {
        "name": "Windows 10 Home",
        "kernel": "10.0.17763.1282 (WinBuild.160101.0800)",
        "build": "17763.1282",
        "platform": "windows",
        "version": "10.0",
        "family": "windows"
      }
    },
    "event": {
      "module": "file_integrity",
      "dataset": "file",
      "kind": "event",
      "category": [
        "file"
      ],
      "type": [
        "change"
      ],
      "action": [
        "updated"
      ]
    },
    "service": {
      "type": "file_integrity"
    },
    "file": {
      "path": "C:\\Windows\\System32\\drivers\\etc\\hosts",
      "type": "file",
      "ctime": "2018-09-15T07:31:36.585Z",
      "size": 806,
      "mtime": "2020-06-19T14:18:05.779Z",
      "inode": "281474976752119",
      "drive_letter": "C",
      "uid": "S-1-5-18",
      "hash": {
        "sha1": "9820fbd5cdae3f54855021cffea9c0716876e8ea"
      },
      "owner": "NT AUTHORITY\\SYSTEM"
    },
    "hash": {
      "sha1": "9820fbd5cdae3f54855021cffea9c0716876e8ea"
    },
    "agent": {
      "hostname": "DESKTOP-[redacted]",
      "ephemeral_id": "d4d06c6e-6bd3-4d4a-a0d3-509a59f9898a",
      "id": "12263842-4868-4cfb-9d96-47d304d4762c",
      "name": "DESKTOP-[redacted]",
      "type": "auditbeat",
      "version": "7.8.0"
    },
    "ecs": {
      "version": "1.5.0"
    }
  },
  "fields": {
    "file.created": [],
    "system.audit.package.installtime": [],
    "process.parent.start": [],
    "event.end": [],
    "file.ctime": [
      "2018-09-15T07:31:36.585Z"
    ],
    "tls.client.not_after": [],
    "system.audit.user.password.last_changed": [],
    "event.ingested": [],
    "@timestamp": [
      "2020-06-19T14:18:05.780Z"
    ],
    "file.mtime": [
      "2020-06-19T14:18:05.779Z"
    ],
    "event.created": [],
    "file.accessed": [],
    "system.audit.host.boottime": [],
    "tls.server.not_before": [],
    "package.installed": [],
    "tls.client.not_before": [],
    "event.start": [],
    "tls.server.not_after": [],
    "process.start": []
  },
  "highlight": {
    "file.path": [
      "@kibana-highlighted-field@C:\\Windows\\System32\\drivers\\etc\\hosts@/kibana-highlighted-field@"
    ],
    "event.category": [
      "@kibana-highlighted-field@file@/kibana-highlighted-field@"
    ],
    "event.type": [
      "@kibana-highlighted-field@change@/kibana-highlighted-field@"
    ],
    "agent.version": [
      "@kibana-highlighted-field@7.8.0@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1592576285780
  ]
}
```

### PR DESCRIPTION
**Pull request by peasead**
## Issues
Resolves #24 

## Summary
The hosts file on endpoints is used to control manual IP to hostname resolutions. The hosts file is the first point of
lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic to
malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or RHEL)
and macOS systems. For Windows systems, this rule requires adding "C:/Windows/System32/drivers/etc" as an additional
path in the "file_integrity" module of auditbeat.yml.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: rta/hosts_file_modify.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    backup = os.path.abspath(hosts_file + '_backup')
+    common.log("Backing up original 'hosts' file.")
+    common.copy_file(hosts_file, backup)
+
+    # add randomness for diffs for FIM module
+    randomness = ''.join(random.sample(ascii_letters, 10))
+    entry = [
+        '',
+        '# RTA hosts_modify was here',
+        '8.8.8.8 https://www.google.com/{random}'.format(random=randomness)
```
Comment (rw-access) [inline rta/hosts_file_modify.py:36]: I think this is a bug: the hosts file expects domains, but this is a URL

Comment (brokensound77) [reply rta/hosts_file_modify.py:36]:
```suggestion
        '8.8.8.8 https://www.{random}.google.com'.format(random=randomness)
```

It is not meant to be a legit connection and so even just adding a comment would suffice.

File: rta/hosts_file_modify.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    backup = os.path.abspath(hosts_file + '_backup')
+    common.log("Backing up original 'hosts' file.")
+    common.copy_file(hosts_file, backup)
+
+    # add randomness for diffs for FIM module
+    randomness = ''.join(random.sample(ascii_letters, 10))
+    entry = [
+        '',
+        '# RTA hosts_modify was here',
+        # '8.8.8.8 https://www.{random}.google.com'.format(random=randomness)
```
Comment (brokensound77) [inline rta/hosts_file_modify.py:36]:
```suggestion
        '# 8.8.8.8 https://www.{random}.google.com'.format(random=randomness)
```

My mistake from earlier

At the moment, the rule is solely looking for modification of the file itself. If down the road we need to emulate an actual processing of a domain lookup defined within, we can revisit

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+updated_date = "2020/07/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+The hosts file on endpoints is used to control manual IP to hostname resolutions. The hosts file is the first point of
+lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic to
+malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or RHEL)
+and macOS systems. For Windows systems, this rule requires adding "C:/Windows/System32/drivers/etc" as an additional
+path in the "file_integrity" module of auditbeat.yml.
```
Comment (threat-punter) [inline rules/cross-platform/impact_hosts_file_modified.toml:14]:
```suggestion
The hosts file on endpoints is used to control manual IP address to hostname resolutions. The hosts file is the first point of
lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic to
malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or RHEL)
and macOS systems. For Windows systems, this rule requires adding "C:/Windows/System32/drivers/etc" as an additional
path in the "file_integrity" module of auditbeat.yml.
```

Comment (threat-punter) [inline rules/cross-platform/impact_hosts_file_modified.toml:14]: Can we include a reference to help people modify their `auditbeat.yml` configuration or is there a level of knowledge that we're assuming that users have?

Comment (peasead) [reply rules/cross-platform/impact_hosts_file_modified.toml]: By default, `auditbeat.yml` has examples in the in `file_integrity` section. 
```
...
# The file integrity module sends events when files are changed (created,
# updated, deleted). The events contain file metadata and hashes.
- module: file_integrity
  paths:
  - /bin
  - /usr/bin
  - /sbin
  - /usr/sbin
  - /etc
...
```

https://www.elastic.co/guide/en/beats/auditbeat/current/auditbeat-reference-yml.html

I'm happy to provide one, but I'm not sure where to put it for reference? Open to suggestions.

Comment (threat-punter) [reply rules/cross-platform/impact_hosts_file_modified.toml]: What do you think about moving the following text from the `description` field to the `notes` field?

```
For Windows systems, this rule requires adding "C:/Windows/System32/drivers/etc" as an additional
path in the "file_integrity" module of auditbeat.yml.
```

Comment (peasead) [reply rules/cross-platform/impact_hosts_file_modified.toml]: Got it. 

Added the note and reference fields.

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+updated_date = "2020/07/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+The hosts file on endpoints is used to control manual IP address to hostname resolutions. The hosts file is the first
+point of lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic
+to malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or
+RHEL) and macOS systems. For Windows systems, this rule requires adding "C:/Windows/System32/drivers/etc" as an
+additional path in the "file_integrity" module of auditbeat.yml.
```
Comment (threat-punter) [inline rules/cross-platform/impact_hosts_file_modified.toml:14]:
```suggestion
RHEL) and macOS systems.
```

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+[metadata]
+creation_date = "2020/07/07"
+ecs_version = ["1.5.0"]
```
Comment (brokensound77) [inline rules/cross-platform/impact_hosts_file_modified.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+note = "For Windows systems, this rule requires adding 'C:/Windows/System32/drivers/etc' as an additional path in the 'file_integrity' module of auditbeat.yml."
+references = ["https://www.elastic.co/guide/en/beats/auditbeat/current/auditbeat-reference-yml.html"]
+risk_score = 47
+rule_id = "9c260313-c811-4ec8-ab89-8f6530e0246c"
+severity = "medium"
+tags = ["Elastic", "Linux", "Windows", "macOS"]
+type = "query"
+
+query = '''
+event.category:file and event.type:change and file.path:(/private/etc/hosts or /etc/hosts or "C:\Windows\System32\drivers\etc\hosts")
```
Comment (brokensound77) [inline rules/cross-platform/impact_hosts_file_modified.toml:28]:
```suggestion
event.category:file and event.type:(creation or change) and file.path:(/private/etc/hosts or /etc/hosts or "C:\Windows\System32\drivers\etc\hosts")
```

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+The hosts file on endpoints is used to control manual IP address to hostname resolutions. The hosts file is the first
+point of lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic
+to malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or
+RHEL) and macOS systems.
+"""
+index = ["auditbeat-*"]
```
Comment (brokensound77) [inline rules/cross-platform/impact_hosts_file_modified.toml:15]: add `logs-endpoint.events.*`

Comment (brokensound77) [reply rules/cross-platform/impact_hosts_file_modified.toml:15]: maybe winlogbeat-* too

**Review by devonakerr — APPROVED** [review]

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+The hosts file on endpoints is used to control manual IP address to hostname resolutions. The hosts file is the first
+point of lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic
+to malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or
+RHEL) and macOS systems.
+"""
+index = ["auditbeat-*", "winlogbeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Hosts File Modified"
+note = "For Windows systems, this rule requires adding 'C:/Windows/System32/drivers/etc' as an additional path in the 'file_integrity' module of auditbeat.yml."
```
Comment (brokensound77) [inline rules/cross-platform/impact_hosts_file_modified.toml:19]:
```suggestion
note = "For Windows systems using auditbeat, this rule requires adding 'C:/Windows/System32/drivers/etc' as an additional path in the 'file_integrity' module of auditbeat.yml."
```

File: rules/cross-platform/impact_hosts_file_modified.toml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+The hosts file on endpoints is used to control manual IP address to hostname resolutions. The hosts file is the first
+point of lookup for DNS hostname resolution so if adversaries can modify the endpoint hosts file, they can route traffic
+to malicious infrastructure. This rule detects modifications to the hosts file on Microsoft Windows, Linux (Ubuntu or
+RHEL) and macOS systems.
+"""
+index = ["auditbeat-*", "winlogbeat-*"]
```
Comment (brokensound77) [inline rules/cross-platform/impact_hosts_file_modified.toml:15]:
```suggestion
index = ["auditbeat-*", "winlogbeat-*", "logs-endpoint.events.*"]
```

This logic would work on the endpoint agent as well

**Review by brokensound77 — APPROVED** [review]
Just the 2 comments, then LGTM 👍

**Pull request merged**